devito/function.py

```diff
@@ -816,61 +778,51 @@ class SparseFunction(CompositeFunction):
 
     def __init__(self, *args, **kwargs):
         if not self._cached():
-            self.nt = kwargs.get('nt', 0)
-            self.npoint = kwargs.get('npoint')
-            kwargs['shape'] = (self.nt, self.npoint)
-            super(SparseFunction, self).__init__(self, *args, **kwargs)
-
-            if self.grid is None:
-                error('SparseFunction objects require a grid parameter.')
-                raise ValueError('No grid provided for SparseFunction.')
-
-            # Allocate and copy coordinate data
-            d = Dimension(name='d')
-            self.coordinates = Function(name='%s_coords' % self.name,
-                                        dimensions=[self.indices[-1], d],
-                                        shape=(self.npoint, self.grid.dim),
-                                        space_order=0)
-            self._children.append(self.coordinates)
-            coordinates = kwargs.get('coordinates', None)
-            if coordinates is not None:
-                self.coordinates.data[:] = coordinates[:]
-
-            # A SparseFunction has no halo region
+            super(SparseFunction, self).__init__(*args, **kwargs)
+
+            npoint = kwargs.get('npoint')
+            if not isinstance(npoint, int) and npoint > 0:
+                raise ValueError('SparseFunction requires parameter `npoint` (> 0)')
+            self.npoint = npoint
+
+            # Grid must be provided
+            grid = kwargs.get('grid')
+            if kwargs.get('grid') is None:
+                raise ValueError('SparseFunction objects require a grid parameter')
+            self.grid = grid
+
+            self.dtype = kwargs.get('dtype', self.grid.dtype)
+            self.space_order = kwargs.get('space_order', 1)
```

Comment:
Would default to 0, `space_order` not really defined for now for Sparse Functions.

Reply:
👍